index.html

```diff
@@ -1,24 +1,32 @@
 <!DOCTYPE html>
 <html lang="en">
-<head>
+  <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Geocoding & search</title>
     <link rel="stylesheet" href="styles.css">
-</head>
-<body>
+  </head>
+  <body>
     <button class="button" id="search-by-location">Search by distance</button>
     <button class="button" id="search-by-price">Search by price</button>
     <button class="button" id="search-by-rating">Search by rating</button>
     <div id="map"></div>
+
+    <input id="filterValue" type="text" />
+    <button id="filterBy">Filter</button>
+
     <div id="restaurant-returned"></div>
+
     <div id="modal-restaurant-info" class="modal">
-        <button class="close-modal" id="close-modal">CLOSE</button>
-        <div id="modal-content"></div>
-    </div> 
-  <script async defer src="https://maps.googleapis.com/maps/api/js?key=AIzaSyBs0DXvW7g88kD3-OS7i4HXHzl_oPAP1LQ&callback=initMap"></script>
-  <script src="map.js"></script>
-</body>
-<script type="text/javascript" src="script.js"></script>
-</html>
\ No newline at end of file
+      <button class="close-modal" id="close-modal">CLOSE</button>
+      <div id="modal-content"></div>
+    </div>
+    <div id="overlay"></div>
+  </body>
+  <script
```

Comment:
Is there a reason why you have removed the "async" and is it still working without this?